src/centralserver/admin-service/src/test/java/org/niis/xroad/centralserver/restapi/openapi/SubsystemsApiControllerTest.java

```diff
@@ -207,16 +210,79 @@ public void shouldFailWhileTryingToAddToRepository() {
+                inOrder.verify(clientIdConverter).convertId(encodedSubsystemId);
+                inOrder.verify(securityServerIdConverter).convertId(encodedSecurityServerId);
+                inOrder.verify(auditData).put(RestApiAuditProperty.SERVER_CODE, securityServerId.getServerCode());
+                inOrder.verify(auditData).put(RestApiAuditProperty.OWNER_CLASS, securityServerId.getOwner().getMemberClass());
+                inOrder.verify(auditData).put(RestApiAuditProperty.OWNER_CODE, securityServerId.getOwner().getMemberCode());
+                inOrder.verify(auditData).put(RestApiAuditProperty.CLIENT_IDENTIFIER, subsystemId);
+                inOrder.verify(subsystemService).unregisterSubsystem(subsystemId, securityServerId);
+            });
+        }
+
+        @ParameterizedTest
+        @ValueSource(strings = {
+                "INVALID-FORMAT",
+                "TEST:CLASS",
+                "TEST:CLASS:CODE"
+        })
+        @DisplayName("Should throw BadRequest if called with invalid subsystem id")
+        void shouldThrowBadRequestExceptionWhenInvalidSubsystemId(String invalidSubsystemId) {
+            Executable testable =
+                    () -> subsystemsApiController.unregisterSubsystem(invalidSubsystemId, encodedSecurityServerId);
+
+            var thrown = assertThrows(BadRequestException.class, testable,
+                    "Expecting BadRequestException when called with invalid subsystem id");
+
+            assertEquals("Invalid subsystem id", thrown.getMessage());
+        }
+
+        @ParameterizedTest
+        @ValueSource(strings = {
+                "INVALID-FORMAT",
+                "TEST:CLASS",
+                "TEST:CLASS:CODE"
+        })
+        @DisplayName("Should throw BadRequest if called with invalid security server id")
+        void shouldThrowBadRequestExceptionWhenInvalidSecurityServerId(String invalidSecurityServerId) {
+            Executable testable =
+                    () -> subsystemsApiController.unregisterSubsystem(encodedSubsystemId, invalidSecurityServerId);
+
+            var thrown = assertThrows(BadRequestException.class, testable,
+                    "Expecting BadRequestException when called with invalid subsystem id");
+
+            assertEquals("Invalid security server id " + invalidSecurityServerId, thrown.getMessage());
+        }
+    }
+
     @Nested
     @DisplayName("deleteSubsystem(String id)")
     class DeleteSubsystem implements WithInOrder {
 
         private final ClientId subsystemClientId = SubsystemId.create("TEST", "CLASS", "MEMBER", "SUBSYSTEM");
-        private final String encodedSubsystemId = "TEST:CLASS:MEMBER:SUBSYSTEM";
+        private final String encodedSubsystemId = subsystemClientId.toShortString(':');
 
         @Test
         @DisplayName("Should delete subsystem")
-        void shouldDeleteSubsystem() {
+        void shouldUnregisterSubsystem() {
```

Comment:
By what method is being called in the body, this should stay called "delete" subsystem